completions/jq

```diff
@@ -34,14 +34,32 @@ _jq()
                 ;;
         esac
 
+    # See whether the output of --help mentions --help
+    SEEMS_TO_HAVE_COMPLETE_HELP=$(jq --help | grep -q -e --help && echo yes || echo no)
+
+    # jq's --help only shows some of its command-line options; some are not
+    # even listed in the man page!
     if [[ $cur == -* ]]; then
-        COMPREPLY=($(compgen -W '$(_parse_help "$1")' -- "$cur"))
+        if [[ $SEEMS_TO_HAVE_COMPLETE_HELP == yes ]]; then
+            # If the output of --help seems complete, use it
+            COMPREPLY=($(compgen -W '$(_parse_help "$1")' -- "$cur"))
+        else
+            # Otherwise, use a hard-coded list of known flags, some of which do
+            # not appear in the output of --help as of jq 1.6.
+            COMPREPLY=($(compgen -W '--version --seq --stream --slurp --raw-input
+                --null-input --compact-input --tab --indent --color-output
+                --monochrome-output --ascii-output --unbuffered --sort-keys
+                --raw-output --join-output --from-file --exit-status
+                --arg --argjson --slurpfile --rawfile --argfile --args --jsonargs
+                --run-tests --help --version' -- "$cur"))
+        fi
         return
     fi
 
     local args
     # TODO: DTRT with args taking 2 options
-    _count_args "" "@(--arg|--arg?(json|file)|--?(slurp|from-)file|--indent|--run-tests|-!(-*)[fL])"
+    # -f|--from-file are not counted here because they supply the filter
```

Comment:
Hm, this commit is already applied as be4e15592a79b6dae173609331fa9a950556bb23, we can drop it from here (so we can eventuall just merge the whole PR instead of having to cherry pick this).